### PR TITLE
CMTOOL-212: Update documentation to include transaction subsystem updates

### PR DESCRIPTION
**Pull request by sgilda**

### CODE REVIEW

File: docs/user-guides/migrations/includes/src/main/resources/topics/ServerMigration-ServerConfiguration-Subsystem-Update-transactions.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+|===
+|Obsolete Attribute
+|Replacement Attribute
+|path
+|object-store-path
+|relative-to
+|object-store-relative-to
+|===
+
+The following attributes are deprecated the *transactions* subsystem. The JBoss Server Migration Tool replaces them with the equivalent {server-target-productName} replacement attributes.
```
Comment (emmartins) [inline docs/user-guides/migrations/includes/src/main/resources/topics/ServerMigration-ServerConfiguration-Subsystem-Update-transactions.adoc]: The tool does not replaces the attributes below

File: docs/user-guides/migrations/includes/src/main/resources/topics/ServerMigration-ServerConfiguration-Subsystem-Update-transactions.adoc [hunk 2]
```diff
@@ -0,0 +1,32 @@
+= Update the transaction Subsystem
+
+The JBoss Server Migration Tool makes the following updates to the *transactions* subsystem configuration to better match the {server-target-productName} default configurations.
```
Comment (emmartins) [inline docs/user-guides/migrations/includes/src/main/resources/topics/ServerMigration-ServerConfiguration-Subsystem-Update-transactions.adoc]: The *transactions* subsystem *XML* configuration updates are necessary for such configurations to be supported by the target server

File: docs/user-guides/migrations/includes/src/main/resources/topics/ServerMigration-ServerConfiguration-Subsystem-Update-transactions.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+The following attributes were removed from the *transactions* subsystem. The JBoss Server Migration Tool replaces them with the equivalent {server-target-productName} attributes.
+
+[cols="50%,50%", options="header"]
+|===
+|Obsolete Attribute
+|Replacement Attribute
+|path
+|object-store-path
```
Comment (emmartins) [inline docs/user-guides/migrations/includes/src/main/resources/topics/ServerMigration-ServerConfiguration-Subsystem-Update-transactions.adoc]: Such attributes are from the management model, yet the changes must be done at XML level, an example:

`
       <subsystem xmlns="urn:jboss:domain:transactions:1.5">
            <core-environment path="eduardo" relative-to="jboss.server.data.dir">
                ...
            </core-environment>
            ...
        </subsystem>
`

becomes 

`<subsystem xmlns="urn:jboss:domain:transactions:4.0">
            <core-environment>
                ...
            </core-environment>
             ...
            <object-store path="eduardo" relative-to="jboss.server.data.dir"/>
        </subsystem>`

**Comment by pkremens** [comment]
Details about xml representation of the model shouldn't be part of the documentation, as this could be changed in the future (that's what are XML schemas and transformers for). From the management api point of view, the `path` was replaced with `object-store-path` and `relative-to` was replaced with `object-store-relative-to`. How the model is represented is in XML is implementation detail which shouldn't be exposed to the user.

```
6.4.x
-----
        <subsystem xmlns="urn:jboss:domain:transactions:1.5">
            <core-environment path="test" relative-to="jboss.server.data.dir">
                <process-id>
                    <uuid/>
                </process-id>
            </core-environment>
            <recovery-environment socket-binding="txn-recovery-environment" status-socket-binding="txn-status-manager"/>
            <coordinator-environment default-timeout="300"/>
        </subsystem>

[standalone@localhost:9999 /] /subsystem=transactions:read-attribute(name=path)
{
    "outcome" => "success",
    "result" => "test"
}
[standalone@localhost:9999 /] /subsystem=transactions:read-attribute(name=relative-to)
{
    "outcome" => "success",
    "result" => "jboss.server.data.dir"
}
------------------------------
After migration to 7.2.0.Beta
------------------------------
        <subsystem xmlns="urn:jboss:domain:transactions:4.0">
            <core-environment>
                <process-id>
                    <uuid/>
                </process-id>
            </core-environment>
            <recovery-environment socket-binding="txn-recovery-environment" status-socket-binding="txn-status-manager"/>
            <coordinator-environment default-timeout="300"/>
            <object-store path="test" relative-to="jboss.server.data.dir"/>
        </subsystem>

[standalone@localhost:9990 /] /subsystem=transactions:read-attribute(name=object-store-path)
{
    "outcome" => "success",
    "result" => "test"
}
[standalone@localhost:9990 /] /subsystem=transactions:read-attribute(name=object-store-relative-to)
{
    "outcome" => "success",
    "result" => "jboss.server.data.dir"
}
```
We should stick with something as simple as:
Removes the `path` and `relative-to` attributes from transaction subsystem and replace them with `object-store-path` reps. `object-store-relative-to`.

**Comment by sgilda** [comment]
@pkremens : Based on @emmartins' feedback, I didn't realize that `object-store-path` and 1object-store-relative-to` were the names when using the CLI. I will update this and the product docs.

**Comment by pkremens** [comment]
@sgilda no problem, these are the names used by any management tool you'd use (web console, management api, ...) that's why it's good to use these

**Comment by emmartins** [comment]
This specific issue occurs and has to be fixed at XML level... Please note that the server fails to start unless the XML is fixed, even on admin mode, pointing to management model attributes is not correct.

**Comment by emmartins** [comment]
Replacing this with #197

**Pull request closed without merging**